pipeline-runner/src/utils.r

```diff
@@ -54,4 +53,33 @@ handle_debug <- function(scdata, config, task_name, sample_id, debug_config) {
         save(seurat_obj, config, task_name, sample_id, num_cells_to_downsample, file = fpath_cont)
         message(sprintf("⚠️ RUN load('%s') to restore environment.", fpath_host))
     }
-}
\ No newline at end of file
+}
+
+
+#' Calculates statistics before/after filter step
+#'
+#' @param seurat_obj \code{SeuratObject}
+#' @param tmp_sample sample name in \code{seurat_obj@orig.ident} to compute statistics for
+#'
+#' @return list with \itemize{
+#'   \item{"num_cells"}{Number of cells in sample}
+#'   \item{"total_genes"}{Number of detected genes in sample}
+#'   \item{"median_genes"}{Median number of genes detected per cell}
+#'   \item{"median_umis"}{Median number of counts per cell}
+#' }
+#' 
+calc_filter_stats <- function(seurat_obj, tmp_sample) {
+  
+  # subset to current sample
+  scdata <- seurat_obj[, seurat_obj$orig.ident == tmp_sample]
```

Comment:
To access the slot `orig.ident` we need to use `@` instead of `$` (`seurat_obj@orig.ident`) . However, to filter by samples I think that is better to use `seurat_obj$samples`.